### ISSUE TITLE
How to disable LZ4 compression in consumer

### ISSUE THREAD

**Issue by archlichking** [tulios/kafkajs#750]
I'm getting this error in kafkajs consumer
```
LZ4 compression not implemented
```
When LZ4 is enabled via following code, kafka is very slow in getting messages
```
const { Kafka, CompressionTypes, CompressionCodecs } = require("kafkajs");
const LZ4 = require("kafkajs-lz4");
CompressionCodecs[CompressionTypes.LZ4] = new LZ4().codec;
```

I tried enable GZIP as https://github.com/tulios/kafkajs/blob/master/docs/Producing.md#-gzip but the same `LZ4 compression not implemented` error is thrown out when I'm running `consumer.run()`.

I'm using `kafkajs@1.12.0`

**Comment by Nevon** [comment]
> I tried enable GZIP as https://github.com/tulios/kafkajs/blob/master/docs/Producing.md#-gzip but the same LZ4 compression not implemented error is thrown out when I'm running consumer.run().

That's probably because you are trying to consume messages that have been produced with LZ4 compression enabled. You won't be able to consume those without having the LZ4 codec enabled. You can verify this by subscribing with `fromBeginning: false` and changing the group id of your consumer group, and then producing messages with gzip compression instead.

Regarding the `kafkajs-lz4` package, we are actually not affiliated with that package, and as far as I know it's pretty much just delegating to `lz4`.

**Comment by alexanderankin** [comment]
fwiw, this just worked for me, thanks for bravely putting an unaffiliated package in the docs.

**Comment by DesmondHsu** [comment]
Can we support skipping message if it cannot be consumed?

**Comment by jxz032** [comment]
Seems neither kafkajs nor kafka-node support LZ4, and need to install another package to decode it. I tried kafkajs-lz4 but it's not working for me. However simply npm install lz4 works for me with custom codec.

const LZ4 = require("lz4");
const LZ4Codec = {
    async compress(encoder) {
        // TODO:
        return LZ4.encode(encoder);
    },

    async decompress(buffer) {
        return LZ4.decode(buffer);
    }
}

CompressionCodecs[CompressionTypes.LZ4] = () => LZ4Codec;

await this.consumer.run({
  eachMessage: async (res) => {
     if(res && res.topic && res.message) {
        var message = res.message;
        var key = message.key;
        var value = message.value;
     }              
  }
})

**Comment by mossc001** [comment]
@jxz032 and @alexanderankin I've been fighting installing and referring the codec for a while and finally given up and am now reaching out to you both please?

I have kafkajs installed in NodeRed and am getting the error below as per the first post; I can't change the compression on Kafka as I don't own it:
`LZ4 compression not implemented`

I've tried installing both "npm install --save kafkajs-lz4" and "npm install lz4" but am struggling to get it referenced in Kafkajs and in which file? How did you both do it?

**Comment by alexanderankin** [comment]
Might need a little more help to reproduce the issue, particularly as I
haven't dealt with this is a while

**Comment by mossc001** [comment]
@alexanderankin , thanks for replying!

I'm running NodeRed and installed Kafkajs via the Palette Manager and ended up with the error `LZ4 compression not implemented` when connecting to a Kafka Producer with LZ4 compression (I've confirmed this with the server owner now).

I then installed kafkajs-lz4 and lz4 using command line using `npm install --save kafkajs-lz4` and `npm install lz4` but the error of `LZ4 compression not implemented` still prevails.

Do I have to reference the library within Kafkajs and modify the node?
